plugin/trino-clickhouse/src/main/java/io/trino/plugin/clickhouse/ClickHouseClient.java

```diff
@@ -842,4 +871,229 @@ private static SliceWriteFunction uuidWriteFunction()
     {
         return (statement, index, value) -> statement.setObject(index, trinoUuidToJavaUuid(value), Types.OTHER);
     }
+
+    // Different versions of ClickHouse may support different min/max values for the
+    // same data type, you can refer to the table below:
+    //
+    // | version | column type | min value           | max value            |
+    // |---------|-------------|---------------------|----------------------|
+    // | any     | UInt8       | 0                   | 255                  |
+    // | any     | UInt16      | 0                   | 65535                |
+    // | any     | UInt32      | 0                   | 4294967295           |
+    // | any     | UInt64      | 0                   | 18446744073709551615 |
+    // | < 21.4  | Date        | 1970-01-01          | 2106-02-07           |
+    // | < 21.4  | DateTime    | 1970-01-01 00:00:00 | 2106-02-06 06:28:15  |
+    // | >= 21.4 | Date        | 1970-01-01          | 2149-06-06           |
+    // | >= 21.4 | DateTime    | 1970-01-01 00:00:00 | 2106-02-07 06:28:15  |
+    //
+    // And when the value written to ClickHouse is out of range, ClickHouse will store
+    // the incorrect result, so we need to check the range of the written value to
+    // prevent ClickHouse from storing the incorrect value.
+    private enum TrinoToClickHouseWriteChecker
+    {
+        UINT8(ImmutableList.of(new LongWriteValueChecker(Predicates.alwaysTrue(), new Range(0L, 255L)))),
+        UINT16(ImmutableList.of(new LongWriteValueChecker(Predicates.alwaysTrue(), new Range(0L, 65535L)))),
+        UINT32(ImmutableList.of(new LongWriteValueChecker(Predicates.alwaysTrue(), new Range(0L, 4294967295L)))),
+        UINT64(ImmutableList.of(new BigDecimalWriteValueChecker(Predicates.alwaysTrue(), new Range(BigDecimal.ZERO, new BigDecimal("18446744073709551615"))))),
+        DATE(ImmutableList.of(
```

Comment:
Should the min/max be extracted as constants so that we can use them in a test which will verify that the range is correct and up-to-date (by directly performing writes via ClickHouse and then reading the results to verify they match).

Reply:
For now we can test min/max with `DataProvider`, which I think is enough (if in the future we find we really need to extract it, we can implement it at that time)

Reply:
As long as we can catch that the supported range in ClickHouse has changed when we update the tested ClickHouse version it should be fine.